test/integration/listener_lds_integration_test.cc

```diff
@@ -409,5 +409,66 @@ TEST_P(ListenerIntegrationTest, MultipleLdsUpdatesSharingListenSocketFactory) {
   }
 }
 
+TEST_P(ListenerIntegrationTest, ChangeListenerAddress) {
+  on_server_init_function_ = [&]() {
+    createLdsStream();
+    sendLdsResponse({MessageUtil::getYamlStringFromMessage(listener_config_)}, "1");
+    createRdsStream(route_table_name_);
+  };
+  initialize();
+  test_server_->waitForCounterGe("listener_manager.lds.update_success", 1);
+  // testing-listener-0 is not initialized as we haven't pushed any RDS yet.
+  EXPECT_EQ(test_server_->server().initManager().state(), Init::Manager::State::Initializing);
+  // Workers not started, the LDS added listener 0 is in active_listeners_ list.
+  EXPECT_EQ(test_server_->server().listenerManager().listeners().size(), 1);
+  registerTestServerPorts({listener_name_});
+
+  const std::string route_config_tmpl = R"EOF(
```

Comment:
nit: the route config can be defined within the http conn manager so we don't use the rds stream here.

Reply:
I think rds could be used here to prevent the listener from completing warming. 

This brings up an interesting question: I'm not sure what the expected semantics are when the new listener is still warming. Should the old listener be guaranteed to still be accepting connections until the new listener is done warming? This is more complicated with the listener address changing. I think it wouldn't be in the case of localhost->wildcard, but unicast->different unicast it could matter.

Reply:
By this being more complicated, do you mean the logic should be different from when a listener is normally updated (without changing the address)? Anyway, from my understanding, the old listener isn't drained until the new one is warmed: https://github.com/envoyproxy/envoy/blob/v1.18.2/source/server/listener_manager_impl.cc#L749